Apps/Src/FaultState.c

```diff
@@ -129,26 +135,52 @@ void EnterFaultState() {
     // TODO: create an interrupt-independent CAN interface, so we can use CAN from within a fault state
     // avoid infinite recursive faults, since our CAN Driver relies on the OS to work
     // also don't call CAN if the watchdog tripped, since CAN won't be initialized
-    /*
-    if ((Fault_BitMap & (Fault_OS | Fault_WDOG)) == 0) {
-        CANData_t data;
-        data.b = 1;
-        CANPayload_t Message;
-        Message.data = data;
-        // Push Trip message to CAN Q
-        CANbus_BlockAndSend(TRIP, Message);
-        // Push Contactor State message to CAN Q
-        data.b = 0;
-        CANbus_BlockAndSend(CONTACTOR_STATE, Message);
-    }
-    */
+
+    //Deinitialize CAN registers
+    CANbus_DeInit();
+    //Reinit CAN in fault state
+    CANbus_Init(false, true);
+    
 
 #ifdef DEBUGMODE
     char command[COMMAND_SIZE];
 #endif
     BSP_WDTimer_Init(); //This is in case we did not pass check contactor and watchdog timer was not initialized
     BSP_WDTimer_Start(); 
     while(1) {
+        //Send Trip Readings
+        CANPayload_t payload;
+        payload.data.w = 1;
+        CANbus_SendMsg_FaultState(TRIP, payload);
+        BSP_PLL_DelayU(MESSAGE_BUFFER);
+
+        //Send Contactor Readings
+        payload.data.b = 0;
+        CANbus_SendMsg_FaultState(CONTACTOR_STATE, payload);
+        BSP_PLL_DelayU(MESSAGE_BUFFER);
+        
+        //Send Current Readings
+        payload.data.w = Amps_GetReading();
+        CANbus_SendMsg_FaultState(CURRENT_DATA, payload);
+        BSP_PLL_DelayU(MESSAGE_BUFFER);
```

Comment:
why is MESSAGE_BUFFER being used to delay micro seconds?
If MESSAGE_BUFFER is being used to delay it should be renamed, otherwise its confusing as it can be interpreted as the size of a message buffer